src/ess.jl

```diff
@@ -197,23 +187,67 @@ end
 
 """
     ess_rhat(
-        samples::AbstractArray{<:Union{Missing,Real},3}; method=ESSMethod(), maxlag=250
+        [estimator,]
+        samples::AbstractArray{<:Union{Missing,Real},3};
+        method=ESSMethod(),
+        split_chains::Int=2,
+        maxlag::Int=250,
     )
 
-Estimate the effective sample size and the potential scale reduction of the `samples` of
-shape `(draws, chains, parameters)` with the `method` and a maximum lag of `maxlag`.
+Estimate the effective sample size and ``\\widehat{R}`` of the `samples` of shape
+`(draws, chains, parameters)` with the `method` and a maximum lag of `maxlag`.
+
+By default, the computed ESS and ``\\widehat{R}`` values correspond to the estimator `mean`.
+Other estimators can be specified by passing a function `estimator` (see below).
+
+`split_chains` indicates the number of chains each chain is split into.
+When `split_chains > 1`, then the diagnostics check for within-chain convergence. When
+`d = mod(draws, split_chains) > 0`, i.e. the chains cannot be evenly split, then 1 draw
+is discarded after each of the first `d` splits within each chain.
+
+For a given estimand, it is recommended that the ESS is at least `100 * chains` and that
+``\\widehat{R} < 1.01``. [^VehtariGelman2021]
+
+See also: [`ESSMethod`](@ref), [`FFTESSMethod`](@ref), [`BDAESSMethod`](@ref),
+[`ess_rhat_bulk`](@ref), [`ess_tail`](@ref), [`rhat_tail`](@ref)
+
+## Estimators
 
-See also: [`ESSMethod`](@ref), [`FFTESSMethod`](@ref), [`BDAESSMethod`](@ref)
+The ESS and ``\\widehat{R}`` values can be computed for the following estimators:
+- `Statistics.mean`
+- `Statistics.median`
+- `Statistics.std`
+- `StatsBase.mad`
+- `Base.Fix2(Statistics.quantile, p::Real)`
+
+[^VehtariGelman2021]: Vehtari, A., Gelman, A., Simpson, D., Carpenter, B., & Bürkner, P. C. (2021).
+    Rank-normalization, folding, and localization: An improved ``\\widehat {R}`` for
+    assessing convergence of MCMC. Bayesian Analysis.
+    doi: [10.1214/20-BA1221](https://doi.org/10.1214/20-BA1221)
+    arXiv: [1903.08008](https://arxiv.org/abs/1903.08008)
 """
+function ess_rhat(samples::AbstractArray{<:Union{Missing,Real},3}; kwargs...)
+    return ess_rhat(Statistics.mean, samples; kwargs...)
+end
+function ess_rhat(f, samples::AbstractArray{<:Union{Missing,Real},3}; kwargs...)
```

Comment:
A flaw of this API is that it doesn't support estimators whose behavior is modified by keyword arguments. e.g. `StatsBase.mad` can take a user-provided `center` keyword, and we could trivially update `_expectand_proxy` to handle that, but there's no corresponding `Base.Fix` wrapper that handles such keywords. Will users need that? Probably not.

Reply:
Agreed, it's inelegant, but this is something we could support in the future. I'd probably wait until a user asks for it though.